sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -953,9 +954,6 @@ pi_result piPlatformGetInfo(pi_platform Platform, pi_platform_info ParamName,
     // information>. Follow the same notation here.
     //
     return ReturnValue(Platform->ZeDriverApiVersion.c_str());
-  default:
-    // TODO: implement other parameters
-    die("Unsupported ParamName in piPlatformGetInfo");
```

Comment:
this change (and other similar) are dangerous since if more values are added to the enum we shall return success for them, which can't be right. Let's move the die after the switch.

Reply:
@smaslov-intel ,  
@sergey-semenov is out for a few weeks. I can make these changes. 

Could I ask for a quick clarification?   Many of the `default: return PI_INVALID_VALUE` clauses were removed with this PR, and you are saying we should restore them, correct? That's easy enough.

But, for this one, the one with the `die()` statement, do you want the `default: die()` restored?  Or something else?  (like returning `PI_INVALID_VALUE`)  ?

Reply:
@cperkinsintel , if restoring "default" is an option then that's what I would suggest to do (it's always a good idea to have a default in my view). If we MUST remove it to silent the warnings then let's move the code in the default after the switch (since all cases return themselves).

In this case, the "die" should be after the switch, but it could actually also be turned into return PI_INVALID_VALUE, which should do the same abort but with a different message.

Reply:
I think, the 'default' statement was removed as dead code causing warnings. 
'default' is considered dead if all enum-values in the switch are listed.

Personally, I like having 'default' statement always letting compiler to remove it if that is dead branch. It seems removing such 'default's is needed here though.

Reply:
That `-Wcovered-switch-default` compile flag is a tough one to work around.  I think we'd all prefer to have the `default` clauses, especially for a plug-in.   We could change the enumerations to have a dummy `END_OF_KNOWN` enumeration value and a convention of never explicitly casing it - using default to catch it.  That would solve both problems.

In the interim, I have re-removed the defaults, and am returning `PI_INVALID_VALUE` with a comment after the switches where that is viable.

Reply:
Can't we just DISABLE the warning in the plugin makefile instead of doing pretty ugly changes?